viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -333,97 +371,82 @@ void SLAMServiceImpl::GetLatestSampledPointCloudMapString(
         }
     }
 
+    // Get data from painted surface in ARGB32 format
     auto painted_surface = painted_slices->surface.get();
+    auto image_format = cairo_image_surface_get_format(painted_surface);
+    if (image_format != cartographer::io::kCairoFormat) {
+        std::string error_log =
+            "Error cairo surface in wrong format, expected Cairo_Format_ARGB32";
+        LOG(ERROR) << error_log;
+        throw std::runtime_error(error_log);
```

Comment:
[q] Does this terminate the cartogropher grpc server?

Reply:
Yes. This function appears 3 times in the code:
- GRPC call to GetPointCloudMap - in a try/catch statement which will stop cartographer if an error occurs via terminate
- CacheMapInLocalizationMode - try/catch statement which will stop cartographer via terminate
- BackupLatestMap - no try/catch.

Question to the team, should this terminate cartographer? And if not should we add a try/catch for BackupLatestMap

Reply:
If we think the format will also fail in future GetPointCloudMap calls if it triggers once then this should terminate cartographer